docs/src/content/en/docs/auth/auth0.mdx

```diff
@@ -0,0 +1,311 @@
+
+### Setting up Auth0 React SDK
+
+First, install and configure the Auth0 React SDK in your application:
+
+```bash copy
+npm install @auth0/auth0-react
+```
+
+```typescript filename="src/auth0-provider.tsx" showLineNumbers copy
+import React from 'react';
+import { Auth0Provider } from '@auth0/auth0-react';
+
+const Auth0ProviderWithHistory = ({ children }) => {
+  const domain = process.env.REACT_APP_AUTH0_DOMAIN;
+  const clientId = process.env.REACT_APP_AUTH0_CLIENT_ID;
+  const audience = process.env.REACT_APP_AUTH0_AUDIENCE;
+
+  return (
+    <Auth0Provider
+      domain={domain}
+      clientId={clientId}
+      authorizationParams={{
+        redirect_uri: window.location.origin,
+        audience: audience,
+        scope: "read:current_user update:current_user_metadata"
+      }}
+    >
+      {children}
+    </Auth0Provider>
+  );
+};
+
+export default Auth0ProviderWithHistory;
+```
+
+### Retrieving access tokens
+
+Use the Auth0 React SDK to authenticate users and retrieve their access tokens:
+
+```typescript filename="lib/auth.ts" showLineNumbers copy
+import { useAuth0 } from '@auth0/auth0-react';
+
+export const useAuth0Token = () => {
+  const { getAccessTokenSilently, loginWithPopup, getAccessTokenWithPopup } = useAuth0();
+
+  const getAccessToken = async () => {
+    try {
+      const token = await getAccessTokenSilently();
+      return token;
+    } catch (error) {
+      // Handle consent or login required errors
+      if (error.error === 'consent_required') {
+        await getAccessTokenWithPopup();
+      } else if (error.error === 'login_required') {
+        await loginWithPopup();
+      }
+      throw error;
+    }
+  };
+
+  return { getAccessToken };
+};
+```
+
+> Refer to the [Auth0 React SDK documentation](https://auth0.com/docs/libraries/auth0-react) for more authentication methods and configuration options.
+
+## Configuring `MastraClient`
+
+When `experimental_auth` is enabled, all requests made with `MastraClient` must include a valid Auth0 access token in the `Authorization` header:
+
+```typescript {7} filename="lib/mastra/mastra-client.ts" showLineNumbers copy
+import { MastraClient } from "@mastra/client-js";
+
+export const createMastraClient = (accessToken: string) => {
+  return new MastraClient({
+    baseUrl: "https://<mastra-api-url>",
+    headers: {
+      Authorization: `Bearer ${accessToken}`
+    }
+  });
+};
+```
+
+> **Note:** The access token must be prefixed with `Bearer` in the Authorization header.
+
+> See [Mastra Client SDK](/docs/server-db/mastra-client.mdx) for more configuration options.
+
+### Making authenticated requests
+
+Once `MastraClient` is configured with the Auth0 access token, you can send authenticated requests:
+
+<Tabs items={["React Component", "curl"]}>
+  <Tab>
+    ```tsx filename="src/components/mastra-api-test.tsx" showLineNumbers copy
```

Comment:
This snippet is too big, any way to minimize it. The React componet in the [Firebase example](https://mastra.ai/en/docs/auth/firebase#making-authenticated-requests) was much smaller